directory_sso_api_client/user.py

```diff
@@ -25,7 +26,17 @@ def get_session_user(self, session_id):
     def get_oauth2_user_profile(self, bearer_token):
         return self.get(
             url=self.endpoints['oauth2_user_profile'],
-            authenticator=BearerAuthenticator(bearer_token)
+            authenticator=AuthenticatorNegotiator(bearer_token=bearer_token)
+        )
+
+    def get_verification_code(self, sso_session_id):
+        authenticator = AuthenticatorNegotiator(
+            bearer_token=None,
+            sso_session_id=sso_session_id,
+        )
```

Comment:
once this is merged we can remove the `bearer_token=None`:

https://github.com/uktrade/directory-client-core/pull/15